packages/grid/_modules_/grid/models/api/componentsApi.ts

```diff
@@ -30,11 +31,27 @@ export interface ApiRefComponentsProperty extends GridIconSlotsComponent {
    * Pagination component rendered in the grid footer by default.
    */
   Pagination: React.ElementType;
+  /**
+   * FilterPanel component rendered when clicking the filter button.
+   */
+  FilterPanel: React.ElementType;
+  /**
+   * ColumnsPanel component rendered when clicking the columns button.
+   */
+  ColumnsPanel: React.ElementType;
+  /**
+   * Panel component wrapping the filters and columns panels.
+   */
+  Panel: React.ElementType;
 }
 
 export interface ComponentsApi {
   /**
    * The set of overridable components used in the grid.
    */
   components: ApiRefComponentsProperty;
+  /**
+   * Overrideable components props dynamic passed to the component at rendering.
+   */
+  componentsProps?: GridSlotsComponentProps;
```

Comment:
Could we update the API pages, e.g. https://material-ui.com/api/data-grid/ to mention it?

Reply:
Yes I will update the doc